src/Service/VendorService/EReolenGlobal/EReolenGlobalVendorService.php

```diff
@@ -0,0 +1,207 @@
+
+        $this->httpClient = $httpClient;
+    }
+
+    /** {@inheritdoc} */
+    public function load(): VendorImportResultMessage
+    {
+        if (!$this->acquireLock()) {
+            return VendorImportResultMessage::error(parent::ERROR_RUNNING);
+        }
+
+        try {
+            $this->progressStart('Starting eReolen Global import from overdrive API');
+
+            $productsUrl = $this->getProductsUrl();
+            $totalCount = $this->limit ?? $this->getTotalItems($productsUrl);
+
+            $batchSize = ($this->limit && $this->limit < self::BATCH_SIZE) ? $this->limit : self::BATCH_SIZE;
+            $offset = 0;
+
+            do {
+                $products = $this->getProducts($productsUrl, $batchSize, $offset);
+
+                $isbnImageUrlArray = [];
+                foreach ($products as $product) {
+                    $coverImageUrl = $product->images->cover->href ?? null;
+
+                    foreach ($product->formats as $format) {
+                        foreach ($format->identifiers as $identifier) {
+                            if (IdentifierType::ISBN === strtolower($identifier->type)) {
+                                $isbnImageUrlArray[$identifier->value] = $coverImageUrl;
+                            }
+                        }
+                    }
+                }
+
+                $this->updateOrInsertMaterials($isbnImageUrlArray, IdentifierType::ISBN);
+
+                $this->progressMessageFormatted($this->totalUpdated, $this->totalInserted, $this->totalIsIdentifiers);
+                $this->progressAdvance();
+
+                $offset += self::BATCH_SIZE;
+            } while ($offset < $totalCount);
+
+            $this->logStatistics();
+
+            $this->progressFinish();
+
+            return VendorImportResultMessage::success($this->totalIsIdentifiers, $this->totalUpdated, $this->totalInserted, $this->totalDeleted);
+        } catch (\Exception $exception) {
+            return VendorImportResultMessage::error($exception->getMessage());
+        }
+    }
+
+    /**
+     * Get products from api.
+     *
+     * @param string $productsUrl
+     * @param int $limit
+     * @param int $offset
+     *
+     * @return array
+     *
+     * @throws IllegalVendorServiceException
+     * @throws UnknownVendorServiceException
+     * @throws GuzzleException
+     */
+    private function getProducts(string $productsUrl, int $limit, int $offset): array
+    {
+        $response = $this->httpClient->request('GET', $productsUrl, [
+            'headers' => [
+                'User-Agent' => $this->getVendor()->getDataServerUser(),
+                'Authorization' => 'Bearer '.$this->getAuthToken(),
+            ],
+            'query' => ['limit' => $limit, 'offset' => $offset],
+        ]);
+
+        $content = $response->getBody()->getContents();
+        $content = json_decode($content);
+
+        return $content->products;
```

Comment:
Don't know if it would be over kill, but could we json_decode this to an array of object `product`. It would make it more type safe.

Reply:
API doesn't provide a library that defines a `Product` object. I think it is overkill for us to implement our own.